Automattic/jetpack
# AI Extension: Start using expandable messages on form creation

This is a proof-of-concept for having a more flexible way of building prompts on the backend that I am tentatively calling "expandable messages", an idea that we got during a Slack discussion: p1692816287680049/1692796253.910939-slack-C054LN8RNVA.

~~Test in pair with D119802-code.~~ The backend change was deployed already.

## Proposed changes:

* Send an expandable message with a form-related type, that will be expanded on the backend to all the necessary messages for creating the form (currently a system and an user message)
* Remove the content of the original prompts, since they are not compatible with this format and would take precedence on the backend

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Go to the block editor of your test site and open the development console
* Add a Jetpack Form block
* Ask for some form (example: "a science fair project submission form")
* Notice the `jetpack-ai-client:ask-question` log message on the console; it should contain just one message:

```
[
    {
        "role": "jetpack-ai",
        "context": {
            "type": "form-ai-extension",
            "content": "",
            "request": "a science fair project submission form"
        }
    }
]
```

* The form should be generated correctly
* All follow up actions on the form should work as well, by sending just the same single message:

```
[
    {
        "role": "jetpack-ai",
        "context": {
            "type": "form-ai-extension",
            "content": "<!-- wp:jetpack/field-name [...]",
            "request": "translate it to portuguese"
        }
    }
]
```
